Assets/Scripts/Mecha/MechaController.cs

```diff
@@ -30,12 +30,16 @@ public class MechaController : MonoBehaviour
     public Animation EnemyCollisionDamageAnimation;
     public Animation WallObstacleDamageAnimation;
 
+    private GameObject lastPlatForm;
```

Comment:
Here, replace with `public GameObject LastPlatform` to create a property.

Also maybe, as commented for the `PlatformMemorizer`, maybe save the coordinates instead of the platform itself.

Reply:
This should be fixed.